app/javascript/components/Greetings.js

```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { fetchGreetings } from '../redux/features/greetings/greetingsSlice';
+
+const Greetings = () => {
+  const { greetings, status } = useSelector((store) => store.greetings)
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchGreetings());
+    console.log(status)
+  }, [dispatch])
+  return (
+    <div>
+        <h1>{greetings[0].message}</h1>
+        <h2>Working</h2>
+    </div>
+  )
```

Comment:
When I run the application nothing is displayed on the webpage. The API endpoint is working properly but there seems to be a problem with fetching data.

I checked the console and I see this error.

![image](https://github.com/Fene-87/hello-rails-react/assets/38088818/90e19475-83d3-4b1f-8dc3-47e16c66c00e)

The error seems to be in the greetings state since it is always undefined which suggests that the API data is not fetched properly.

![image](https://github.com/Fene-87/hello-rails-react/assets/38088818/9fd846dd-60ca-4df7-9db1-b4e4f16e2539)

- Kindly debug this issue and make sure that the correct greeting state is displayed.

Reply:
Hey @KDlamini Thanks for your feedback. I have finally managed to fix it.